src/python/deepgnn/graph_engine/snark/converter/process.py

```diff
@@ -83,21 +84,23 @@ def converter_process(
         if line == FLAG_ALL_DONE:
             break
 
-        node = decoder.decode(line)  # type: ignore
-        writer.add(node)
-        node_alias.add(node)
-
-        for eet in node["edge"]:
-            edge_alias.add(eet)
-            edge_weight[eet["edge_type"]] += eet["weight"]
-            edge_type_count[eet["edge_type"]] += 1
-
-        edge_count += len(node["edge"])
-        node_count += 1
-        node_weight[node["node_type"]] += float(node["node_weight"])
-        node_type_count[node["node_type"]] += 1
-
-    writer.close()
+        for src, dst, typ, weight, features in decoder.decode(line):  # type: ignore
+            if src == -1:
+                node_writer.add(dst, typ, features)
+                edge_writer.add_node()
+                node_alias.add(dst, typ, weight)
+                node_weight[typ] += float(weight)
+                node_type_count[typ] += 1
+                node_count += 1
+            else:
+                edge_writer.add(dst, typ, weight, features)
```

Comment:
where do we verify edges are sorted by dst? without ordering the output of edge writer will be garbage. Currently it is done in NodeWriter, it records an offset to all edges belonging to it. Here you take lines from a queue and ordering is not guaranteed: a node data might be sent to worker processing partition 1, but some of its edges might go to a worker assigned a partition 2.